subworkflows/local/bam_to_fastq.nf

```diff
@@ -0,0 +1,32 @@
+include { SAMTOOLS_FASTQ } from '../../modules/nf-core/samtools/fastq/main'
+
+workflow BAM_TO_FASTQ {
+
+    take:
+    ch_sample // channel: [ val(meta), reads ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Filter out BAM from fastq
+    ch_sample
+        .map { meta, fastq -> [ meta + [ 'single_end': true ], fastq ] }
+        .branch { meta, reads ->
+            fastq: reads.extension == 'gz'
+            bam: reads.extension == 'bam'
+        }
+        .set { ch_filetypes }
+
+    ch_filetypes.fastq.set { ch_sample }
+
+    SAMTOOLS_FASTQ ( ch_filetypes.bam, false )
```

Comment:
Was some time since I ran bam to fastq conversions but then it was recommended to do  `samtools collate` before running `samtools fastq`. I guess that it shouldn't be necessary for uBAMs.

Reply:
I think it's only necessary for short reads? There should never be multiple reads with the same name.